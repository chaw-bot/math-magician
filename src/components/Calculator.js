import React from 'react';
import calculate from '../logic/calculator';
import operate from '../logic/operate';
class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="Calc-body">
        <div className="Calc-display">0</div>
        <div className="Calc-btns">
          <div className="Grey-btns">
            <div className="Grey-btns1">
              <div className="Calc-btn">AC</div>
              <div className="Calc-btn">+/-</div>
              <div className="Calc-btn">%</div>
              <div className="Calc-btn">7</div>
              <div className="Calc-btn">8</div>
              <div className="Calc-btn">9</div>
              <div className="Calc-btn">4</div>
              <div className="Calc-btn">5</div>
              <div className="Calc-btn">6</div>
              <div className="Calc-btn">1</div>
              <div className="Calc-btn">2</div>
              <div className="Calc-btn">3</div>
            </div>
            <div className="Grey-btns2">
              <div className="Calc-btn">0</div>
              <div className="Calc-btn">.</div>
            </div>
          </div>

          <div className="Orange-btns">
            <div className="Calc-btn">รท</div>
            <div className="Calc-btn">x</div>
            <div className="Calc-btn">-</div>
            <div className="Calc-btn">+</div>
            <div className="Calc-btn">=</div>
          </div>
        </div>
      </div>
    );
  }
}

export default Calculator;
