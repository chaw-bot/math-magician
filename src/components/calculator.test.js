import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('Render calculator', () => {
  const tree = renderer
    .create(<Calculator />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
