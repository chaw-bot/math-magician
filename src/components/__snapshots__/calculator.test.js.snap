// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render calculator 1`] = `
<div
  className="calc-page"
>
  <div>
    <h2
      className="title2"
    >
      Let's do some math!
    </h2>
  </div>
  <div
    className="Calc-body"
  >
    <div
      className="Calc-display"
    >
      0
    </div>
    <div
      className="Calc-btns"
    >
      <div
        className="Grey-btns"
      >
        <div
          className="Grey-btns1"
        >
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </div>
        <div
          className="Grey-btns2"
        >
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="Calc-btn"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </div>
      </div>
      <div
        className="Orange-btns"
      >
        <button
          className="Calc-btn"
          onClick={[Function]}
          type="button"
        >
          รท
        </button>
        <button
          className="Calc-btn"
          onClick={[Function]}
          type="button"
        >
          x
        </button>
        <button
          className="Calc-btn"
          onClick={[Function]}
          type="button"
        >
          -
        </button>
        <button
          className="Calc-btn"
          onClick={[Function]}
          type="button"
        >
          +
        </button>
        <button
          className="Calc-btn"
          onClick={[Function]}
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</div>
`;
